### PR TITLE
Fix resource leaks.

### PR DESCRIPTION
**Pull request by monaka**
### What does this PR do?

Fix resource leaks (Not calling `close()`) detected by Eclipse/JDT.
They should be fixed even though they don't crash Che immediately.

### CODE REVIEW

**Comment by skabashnyuk** [comment]
ci-test

File: core/commons/che-core-commons-lang/src/main/java/org/eclipse/che/commons/lang/TarUtils.java [hunk 1]
```diff
@@ -161,26 +161,27 @@ public static void untar(File tar, File targetDir) throws IOException {
...
-      if (tarEntry.isDirectory()) {
-        if (!file.mkdirs()) {
-          throw new IOException("Unable to create folder " + file.getAbsolutePath());
-        }
-      } else {
-        final File parent = file.getParentFile();
-        if (!parent.exists()) {
-          if (!parent.mkdirs()) {
-            throw new IOException("Unable to create folder " + parent.getAbsolutePath());
+    try (TarArchiveInputStream tarIn = new TarArchiveInputStream(in)) {
```
Comment (skabashnyuk) [inline core/commons/che-core-commons-lang/src/main/java/org/eclipse/che/commons/lang/TarUtils.java]: I think the idea might be in a fact that caller of untar method suppose to close ```InputStream``` for example https://github.com/eclipse/che/blob/master/core/commons/che-core-commons-lang/src/main/java/org/eclipse/che/commons/lang/TarUtils.java#L158-L160

**Comment by skabashnyuk** [comment]
ci-test

**Comment by monaka** [comment]
I run "mvn fmt:format" and force-pushed here.

**Comment by skabashnyuk** [comment]
ci-test

**Comment by monaka** [comment]
ci-test

**Comment by sleshchenko** [comment]
The tests fail to start because of an issue in Theia module. I've registered an issue to fix it https://github.com/eclipse/che/issues/12452

**Comment by benoitf** [comment]
ci-build

**Comment by benoitf** [comment]
ci-test

**Review by skabashnyuk — APPROVED** [review]

**Comment by skabashnyuk** [comment]
@eclipse/eclipse-che-qa can you take a look?

**Review by garagatyi — APPROVED** [review]

**Comment by artaleks9** [comment]
Selenium tests execution on Eclipse Che Multiuser on OCP (https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/1471/) doesn't show any regression against this Pull Request.

**Pull request merged**